### PR TITLE
Reworked Makefiles to use pattern rules

### PR DESCRIPTION
**Pull request by edwarnicke**

### CODE REVIEW

File: .nsm.mk [hunk 1]
```diff
@@ -32,60 +32,12 @@ DOCKER_SRIOV_CONTROLLER=networkservicemesh/sriov-controller
...
-	@if [ "x${COMMIT}" != "x" ] ; then \
-		docker tag ${DOCKER_SIDECAR_INJECTOR} ${DOCKER_SIDECAR_INJECTOR}:${COMMIT} ;\
-	fi
-
-.PHONY: docker-build-sriov-controller
-docker-build-sriov-controller: docker-build-release
-	@${DOCKERBUILD} -t ${DOCKER_SRIOV_CONTROLLER}  -f build/Dockerfile.sriov-controller .
-	@if [ "x${COMMIT}" != "x" ] ; then \
-		docker tag ${DOCKER_SRIOV_CONTROLLER} ${DOCKER_SRIOV_CONTROLLER}:${COMMIT} ;\
+		docker tag ${ORG}/release ${ORG}/release:${COMMIT} ;\
```
Comment (fkautz) [inline .nsm.mk]: Is the back slack required?

File: .nsm.mk [hunk 2]
```diff
@@ -16,14 +16,14 @@
...
-DOCKER_RELEASE=networkservicemesh/release
-DOCKER_SIDECAR_INJECTOR=networkservicemesh/sidecar-injector
-DOCKER_SRIOV_CONTROLLER=networkservicemesh/sriov-controller
+ORG=networkservicemesh
+
+.PHONY: docker-build-%
+docker-build-%: docker-build-release
+	@${DOCKERBUILD} -t ${ORG}/$* -f build/Dockerfile.$* .
+	@if [ "x${COMMIT}" != "x" ] ; then \
+		docker tag ${ORG}/$* ${ORG}/$*:${COMMIT} ;\
```
Comment (fkautz) [inline .nsm.mk]: See line 40

File: .nsm.mk (hunk 1 above)
Comment (edwarnicke) [reply .nsm.mk]: In multiline make rules, yes.

File: .nsm.mk (hunk 2 above)
Comment (edwarnicke) [reply .nsm.mk]: See response to line 40 :)  Also... I found it that way :)

**Pull request merged**